Override default value of retryhttp 429 for prometheus remote write to true in static mode (#4901)
grafana/agent: docs/sources/static/release-notes.md

[release-notes-flow]: "/docs/grafana-cloud/ -> ../flow/release-notes"
{{% /docs/reference %}}

> ## v0.36

Reviewer: We cut the RC already, so this will need an update.

Author: so v0.37?

Reviewer: yes

> ### Breaking change: The prometheus default value of `retry_on_http_429` is overriden to `true` for the `queue_config` in `remote_write`

Reviewer: Why "prometheus default value"?

Author: because the default value is taken from prometheus (https://prometheus.io/docs/prometheus/2.42/configuration/configuration/#remote_write). This is the page which is referred to in the static documentation of the agent. This might actually be confusing for the users that we refer to this page which clearly says that the retry default value is false but in the agent it will be actually true

Reviewer: I see. I find the phrasing confusing though. I think we should just say:

> Breaking change: The default value of `retry_on_http_429` is overriden to `true` for the `queue_config` in `remote_write` in `metrics` config.

And then add a note saying something like:
> NOTE: the default set by Grafana Agent Static Mode is different to the default set by Prometheus.

Finally, shouldn't this be mentioned somewhere in the static mode docs? What do you think?

Author: I extended the note in the static doc

[release-notes-flow]: "/docs/grafana-cloud/ -> ../flow/release-notes"
{{% /docs/reference %}}

## v0.37

### Breaking change: The default value of `retry_on_http_429` is overriden to `true` for the `queue_config` in `remote_write` in `metrics` config.

> > **Note:** The default set by Grafana Agent Static Mode is different to the default set by Prometheus.

Reviewer 2: ```suggestion
> **Note:** The default set by Grafana Agent Static Mode is different than the default set by Prometheus.
```

> > **Note:** The default set by Grafana Agent Static Mode is different than the default set by Prometheus.

Reviewer 2: ```suggestion
{{% admonition type="note" %}}
The default set by Grafana Agent Static Mode is different than the default set by Prometheus.
{{% /admonition %}}
```

Just converting note to admonition syntax

> The prometheus default value of `retry_on_http_429` is overriden to `true` for the `queue_config` in `remote_write`
> so that the agent can retry sending and avoid data being lost for metric pipelines by default.

Reviewer 2: ```suggestion
The Prometheus default value for `retry_on_http_429` is set to `true` for the `queue_config` in `remote_write`.
This changed default setting allows the agent to retry sending data when it receives an HTTP 429 error and helps avoid losing data in metric pipelines.
```

Could we change the wording a little here?

* If you set the `retry_on_http_429` explicitly - no action is required.

> * If you do not set `retry_on_http_429` explicitly and you do *not* want to retry on HTTP 429, make sure you set it to `false` as you upgrade to this new version.

Reviewer 2: ```suggestion
* If you do not set `retry_on_http_429` explicitly and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.
```

> * If you set the `retry_on_http_429` explicitly - no action is required.
> * If you do not set `retry_on_http_429` explicitly and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.

Reviewer 2: ```suggestion
* If you explicitly set the `retry_on_http_429`, no action is required.
* If you do not explicitly set `retry_on_http_429` and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.
```
